### ISSUE TITLE
Agent side of conversation vanishing

### ISSUE THREAD

**Issue by ScotTFO** [langgenius/dify#11408]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.13.1

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

1. Create Agent
2. Discuss with the agent
3. Move to a different conversation
4. Move back to initial conversation

### ✔️ Expected Behavior

The full conversation should exist between the user and agent 

### ❌ Actual Behavior

The agent side of the conversation is vanishing when revisiting.
![image](https://github.com/user-attachments/assets/3fcc4521-8266-49b0-b22e-c85dd57f1600)

**Comment by ScotTFO** [comment]
@dosu that is not the same issue

**Comment by crazywoola** [comment]
Before step 3, did the conversation finish generating things?

**Comment by ScotTFO** [comment]
> Before step 3, did the conversation finish generating things?

Yes it did.  I have other full conversations with about 20 back and forths and the same thing happened.

**Comment by crazywoola** [comment]
> > Before step 3, did the conversation finish generating things?
> 
> Yes it did. I have other full conversations with about 20 back and forths and the same thing happened.

Can you paste the logs of the chat bubble, you can hover the mouse on it and get the output log there.

**Comment by zolgear** [comment]
Switching the conversation during the output of a lengthy source code response reproduces the issue.

**Comment by ScotTFO** [comment]
> > > Before step 3, did the conversation finish generating things?
> > 
> > 
> > Yes it did. I have other full conversations with about 20 back and forths and the same thing happened.
> 
> Can you paste the logs of the chat bubble, you can hover the mouse on it and get the output log there.

![image](https://github.com/user-attachments/assets/c2a5393b-f18d-4653-bc20-684b593df151)

**Comment by ScotTFO** [comment]
> Switching the conversation during the output of a lengthy source code response reproduces the issue.
> 
> However, this behavior appears to be the same as explicitly pressing "Stop responding," so it seems to be by design.

I will check and see if that is the case for me as well, but in this instance, I did allow the text to finish generating.  I was when I left the conversation and later came back to it that the replies were then missing.

**Comment by ScotTFO** [comment]
I did a bit more testing, and as far as I can tell, this behavior is specific to using the o1-preview model for OpenAI.  When I switched to GPT-4o, it was working as expected.

**Comment by sofs2005** [comment]
this issue happened very often after I upgraded to 0.13.1, and I'm using gpt-40-mini.
in history I can see the reply, but front end just empty.

**Comment by sofs2005** [comment]
> this issue happened very often after I upgraded to 0.13.1, and I'm using gpt-40-mini. in history I can see the reply, but front end just empty.

after I reversed to 0.13.0, it's better

**Comment by zolgear** [comment]
The issue was reproduced with the agent (default) + GPT-4o-mini (Azure).

From the nginx logs and browser requests, it appears that a stop was requested even though the conversation had already been completed.

Since the data is stored in the database, switching the conversation twice caused it to display properly.
